src/context/ProfileContext.js

```diff
@@ -0,0 +1,32 @@
+import { useState, useContext, createContext, useEffect } from 'react';
+import { getProfile } from '../services/profiles';
+
+const ProfileContext = createContext();
```

Comment:
Context shouldn't really be necessary for the profile as it is not being used throughout the entire application. A custom hook handling this state that is called as needed may make more sense and avoid some of the bugs that come with your useEffect running before a user has a profile